### PR TITLE
J111 notifications page

### PR DESCRIPTION
**Pull request by EugeniusGR**
image

and read notification:
image

### CODE REVIEW

File: frontend/src/components/common/header/header.tsx [hunk 1]
```diff
@@ -51,8 +52,15 @@ const Header: React.FC = () => {
...
-                <Button label="+ Create Podcast" type={ButtonType.BUTTON} />
+              <Button
+                label="+ Create Podcast"
+                href={AppRoute.PODCASTS_EDIT}
+                type={ButtonType.BUTTON}
+              />
+              <Link to={AppRoute.NOTIFICATIONS}>
+                <button className={styles.notification}>
+                  <img src={headerBell} width="23" height="23" />
+                </button>
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
                  <img src={headerBell} width="23" height="23" loading="lazy" alt="" />
```

File: frontend/src/components/common/header/styles.module.scss [hunk 2]
```diff
@@ -49,6 +49,17 @@
...
 
+.notification {
+  position: relative;
+  width: 23px;
+  padding: 0;
+  border: none;
+  background-color: transparent;
+  height: 23px;
+  margin-left: 42px;
+  cursor: pointer;
```
Comment (ghost) [inline frontend/src/components/common/header/styles.module.scss]:
```suggestion
```

File: frontend/src/components/notifications/components/notification-item.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import defaultLogo from 'assets/img/default-notifications.png';
```
Comment (ghost) [inline frontend/src/components/notifications/components/notification-item.tsx]:
```suggestion
import notification-icon from 'assets/img/notification-icon.png';
```

File: frontend/src/components/notifications/components/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  .notificationDate {
+    margin-top: 50px;
+    color: #8b8b8b;
+  }
+}
+
+.containerChecked {
+  @extend .container;
```
Comment (ghost) [inline frontend/src/components/notifications/components/styles.module.scss]:
```suggestion
  .container;
```
don't use extend

**Review by ghost — CHANGES_REQUESTED** [review]
image
use our common checkbox component, and use our primary color for the eye icon & see messages button

File: frontend/src/components/common/header/header.tsx [hunk 5]
```diff
@@ -51,8 +52,18 @@ const Header: React.FC = () => {
...
+              <Button
+                label="+ Create Podcast"
+                href={AppRoute.PODCASTS_EDIT}
+                type={ButtonType.BUTTON}
+              />
+              <Link to={AppRoute.NOTIFICATIONS}>
+                <img
+                  src={headerBell}
+                  width="23"
+                  height="23"
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx:62]:
```suggestion
                  width="23"
                  height="23"
                  loading="lazy"
                  alt=""
```

File: frontend/src/components/notifications/components/notification-item.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import notificationIcon from 'assets/img/notification-icon.png';
+import newNotificationEye from 'assets/img/eye.svg';
+import { AppRoute } from 'common/enums/enums';
+import { Link } from 'components/common/common';
+import styles from './styles.module.scss';
+
+const NotificationItem: React.FC = () => {
+  return (
+    <div className={styles.container}>
```
Comment (ghost) [inline frontend/src/components/notifications/components/notification-item.tsx]:
```suggestion
    <li className={styles.container}>
```

File: frontend/src/components/notifications/notifications.tsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import NotificationItem from './components/notification-item';
+import styles from './styles.module.scss';
+
+const Notifications: React.FC = () => {
+  return (
+    <div className={styles.container}>
+      <NotificationItem />
+    </div>
```
Comment (ghost) [inline frontend/src/components/notifications/notifications.tsx]:
```suggestion
       <ul className={styles.container}>
         <NotificationItem />
        </ul>
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/header/header.tsx [hunk 8]
```diff
@@ -50,8 +51,20 @@ const Header: React.FC = () => {
...
+                  loading="lazy"
+                  alt="notifications"
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
                  alt=""
```

File: frontend/src/components/notifications/notifications.tsx [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import NotificationItem from './components/notification-item';
+import styles from './styles.module.scss';
+
+const Notifications: React.FC = () => {
+  return (
+    <li className={styles.container}>
+      <NotificationItem />
+    </li>
```
Comment (ZubkoWhatislove) [inline frontend/src/components/notifications/notifications.tsx:8]:
```suggestion
    <ul className={styles.container}>
      <NotificationItem />
    </ul>
```

File: frontend/src/components/notifications/components/notification-item.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+          See message
+        </Link>
+      </div>
+      <div className={styles.notificationStatus}>
+        <button className={styles.buttonEye}>
+          <img src={newNotificationEye} width="25" height="25" />
+        </button>
+        <span className={styles.notificationDate}>09:31</span>
+      </div>
+    </ul>
```
Comment (ghost) [inline frontend/src/components/notifications/components/notification-item.tsx]:
```suggestion
    <li className={styles.container}>
      <div className={styles.notificationLogo}>
        <img
          className={styles.image}
          src={notificationIcon}
          width="40"
          height="40"
        />
      </div>
      <div className={styles.notificationInfo}>
        <h2 className={styles.infoTitle}>
          You have a new message from Lincoln
        </h2>
        <p className={styles.notificationText}>
          Four score and seven years ago our fathers brought forth, on this
          continent, a new nation, conceived in Liberty, and dedicated to the
          proposition that all men are created equal. Now we are engaged in a
          great civil war, testing whether that nation, or any nation so
          conceived, and so dedicated, can long endure.
        </p>
        <Link to={AppRoute.NOTIFICATIONS} className={styles.link}>
          See message
        </Link>
      </div>
      <div className={styles.notificationStatus}>
        <button className={styles.buttonEye}>
          <img src={newNotificationEye} width="25" height="25" />
        </button>
        <span className={styles.notificationDate}>09:31</span>
      </div>
    </li>
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — COMMENTED** [review]
add `notifications` reducer

File: frontend/src/components/notifications/notifications.tsx [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+  );
+};
```
Comment (ghost) [inline frontend/src/components/notifications/notifications.tsx]:
```suggestion
const Notifications: React.FC = () => {
  const { notifications } = useSelector(({ notifications }) => ({
    notifications: notifications.notifications
  }));
  return (
    <li className={styles.container}>
    {notifications.map((it) => (
           <NotificationItem notification={it} key={it.id} />
    ))}
    </li>
  );
};
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by JuliaDzyuba — DISMISSED** [review]

**Comment by EugeniusGR** [comment]
I created just mocked reducer coz I have troubles with backend part

File: frontend/src/store/notification/reducer.ts [hunk 12]
```diff
@@ -0,0 +1,29 @@
+import { createReducer } from '@reduxjs/toolkit';
+import { DataStatus } from 'common/enums/enums';
+import { UserNotification } from 'common/types/types';
+import { loadNotification } from './actions';
+
+type State = {
+  dataStatus: DataStatus;
+  notifications: UserNotification[] | UserNotification | null;
```
Comment (ghost) [inline frontend/src/store/notification/reducer.ts]:
```suggestion
  notifications: UserNotification[];
```

File: frontend/src/store/notification/common.ts [hunk 13]
```diff
@@ -0,0 +1,6 @@
+enum ActionType {
+  LOAD_NOTIFICATION = 'notification/load-notifications',
+  LOAD_USER = 'user-profile/load-user',
```
Comment (ghost) [inline frontend/src/store/notification/common.ts]:
```suggestion
```

File: frontend/src/store/notification/actions.ts [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+import { AsyncThunkConfig, User } from 'common/types/types';
+import { ActionType } from './common';
+
+const loadNotification = createAsyncThunk<User, number, AsyncThunkConfig>(
+  ActionType.LOAD_NOTIFICATION,
+  async (id, { extra }) => {
+    const { userApi } = extra;
+    const user = await userApi.getById(id);
+
+    return user;
```
Comment (ghost) [inline frontend/src/store/notification/actions.ts]: something wrong

**Review by ghost — CHANGES_REQUESTED** [review]

File: shared/src/common/types/notification/notifications.type.ts [hunk 15]
```diff
@@ -0,0 +1,9 @@
+type Notifications = {
```
Comment (ghost) [inline shared/src/common/types/notification/notifications.type.ts]:
```suggestion
type Notification = {
```

Comment (ghost) [reply shared/src/common/types/notification/notifications.type.ts]: don't forget to rename file

File: shared/src/common/enums/api/notification-api-path.enum.ts [hunk 16]
```diff
@@ -0,0 +1,5 @@
+declare enum NotificationApiPath {
```
Comment (ghost) [inline shared/src/common/enums/api/notification-api-path.enum.ts]:
```suggestion
 enum NotificationsApiPath {
```

Comment (ghost) [reply shared/src/common/enums/api/notification-api-path.enum.ts]: don't forget about filename

File: shared/src/common/types/user-notification/user-notification.type.ts [hunk 17]
```diff
@@ -1,10 +1,12 @@
 import { UserNotificationStatus } from '~/common/enums/enums';
+import { Notifications } from 'src/common/types/notification/notifications.type';
 
 type UserNotification = {
   id: number;
   userId: number;
   notificationId: number;
   status: UserNotificationStatus;
+  notifications : Notifications;
```
Comment (ghost) [inline shared/src/common/types/user-notification/user-notification.type.ts]:
```suggestion
  notifications : Notification[];
```

File: shared/src/common/types/user-notification/user-notification.type.ts [hunk 18]
```diff
@@ -1,10 +1,12 @@
 import { UserNotificationStatus } from '~/common/enums/enums';
+import { Notifications } from 'src/common/types/notification/notifications.type';
```
Comment (ghost) [inline shared/src/common/types/user-notification/user-notification.type.ts]:
```suggestion
import { Notifications } from '~/common/types/notification/notifications.type';
```

File: frontend/src/components/notifications/components/notification-item.tsx [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+        <p className={styles.notificationText}>
+          {notification.notifications.message}
+        </p>
+        <Link to={AppRoute.NOTIFICATIONS} className={styles.link}>
+          See message
+        </Link>
+      </div>
+      <div className={styles.notificationStatus}>
+        <button className={styles.buttonEye}>
+          <img src={newNotificationEye} width="25" height="25" />
```
Comment (ghost) [inline frontend/src/components/notifications/components/notification-item.tsx]: add this image as background image

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by yevhenprybytok — DISMISSED** [review]

**Comment by EugeniusGR** [comment]
[Notifications page](https://trello.com/c/Rrz6mZbX/111-notifications-page)

File: frontend/src/components/notifications/notifications.tsx [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+
+const Notifications: React.FC = () => {
+  const { notifications } = useAppSelector(({ notification }: RootState) => ({
+    notifications: notification.notifications,
+  }));
+
+  return (
+    <ul className={styles.container}>
+      {notifications.map((notification) => {
+        <NotificationItem notification={notification} />;
```
Comment (what1s1ove) [inline frontend/src/components/notifications/notifications.tsx]: add key

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**